fix(Core/Spells): Improve teleport destination for a certain class of spell

EricksOliveira opened the pull request:
Improve teleport destination for a certain class of spell. The implementation of Spell::SelectImplicitTargetDestTargets to be more specific.

<!-- First of all, THANK YOU for your contribution. -->

<!-- How to title your Pull Request, Description, Co-Authors (Cherry Pick) and others, please see the link below -->
<!-- https://www.azerothcore.org/wiki/commit-message-guidelines -->

## Changes Proposed:
<!-- If your pull request promotes complex changes that require a detailed explanation, please describe them in detail specifying what your solution is and what is it meant to address. -->
This PR proposes changes to:
-  [X] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:
<!-- If your fix has a relating issue, link it below -->
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/19859
- Closes https://github.com/chromiecraft/chromiecraft/issues/8669

## SOURCE:
<!-- If you can, include a source that can strengthen your claim -->
The changes have been validated through:
- [X] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [X] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:
<!-- Does it build without errors? Did you test in-game? What did you test? On which OS did you test? Describe any other tests performed -->
This PR has been:
- [X] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.


## How to Test the Changes:
<!-- Describe in a detailed step-by-step order how to test the changes -->

- [X] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:
<!-- Is there anything else left to do after this PR? -->

- [ ]
- [ ]

<!-- If you intend to contribute repeatedly to our project, it is a good idea to join our discord channel. We set ranks for our contributors and give them access to special resources or knowledge: https://discord.com/invite/GyFvXpk7)
     Do not remove the instructions below about testing, they will help users to test your PR -->
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

EricksOliveira committed:
Fix Killing Spree teleport and melee positioning

Fix incorrect hitbox values for Supremus (DisplayID 21145) in creature_model_info.

The BoundingRadius and CombatReach were set to extreme values, causing invalid teleport destinations for abilities that rely on melee positioning (such as Killing Spree and Shadowstep), resulting in attacks hitting the air or teleporting the Rogue far away from the target.

Values were adjusted to match bosses of similar size (e.g. Archimonde), restoring correct blizzlike behavior without any core or spell modifications.

EricksOliveira committed:
fix

EricksOliveira committed:
Fix Rogue Killing Spree teleport distance and positioning

EricksOliveira committed:
Fix back-teleport positioning for Killing Spree, Shadowstep and Blink

Nyeriah commented:
This is starting to sound like AI slop. I have to ask, was AI involved in this fix entirely or partially?

EricksOliveira commented:
Partial, for testing purposes. Currently, there is no general fix. Only applies to each specific spell.

EricksOliveira committed:
Improve teleport destination for a certain class of spell

Improve teleport destination for a certain class of spell. The implementation of Spell::SelectImplicitTargetDestTargets to be more specific.

EricksOliveira committed:
.

EricksOliveira committed:
Merge branch 'master' into Killing_Spree
